util/efd_sched.py

```diff
@@ -0,0 +1,231 @@
+    `LSST_DISABLE_BUCKET_VALIDATION` to "1".
+    """
+
+    os.environ["LSST_DISABLE_BUCKET_VALIDATION"] = "1"
+    scheduler_resource_path = ResourcePath(scheduler_url.replace("https://s3.cp.lsst.org/", "s3://rubin:"))
+    scheduler_pickle_bytes = scheduler_resource_path.read()
+
+    if destination is None:
+        result = pickle.loads(scheduler_pickle_bytes)
+    else:
+        if os.path.isdir(destination):
+            base_file_name = (
+                scheduler_resource_path.basename()
+                .replace("Scheduler:2_Scheduler:2", "auxtel")
+                .replace(":", "")
+                + ".xz"
+            )
+            file_name = os.path.join(destination, base_file_name)
+        else:
+            file_name = destination
+
+        this_open = lzma.open if file_name.endswith(".xz") else open
+        with this_open(file_name, "wb") as this_file:
+            this_file.write(scheduler_pickle_bytes)
+
+        result = file_name
+
+    return result
+
+
+async def get_scheduler_at_time(desired_time, destination=None, efd="usdf_efd"):
+    """Get the scheduler, conditions tuple for a desired time.
+
+    Parameters
+    ----------
+    desired_time : `str`
+        The desired time to get the scheduler for in iso8601 format.
+    destination : `str`, optional
+        The file path to save the scheduler to.
+        If None, the (scheduler, conditions) tuple is returned.
+    efd : `str`, optional
+        The EFD to query for schedulers. Defaults to 'usdf_efd'.
+
+    Returns
+    -------
+    result : `tuple` or `str`
+        If `destination` is None, the loaded (scheduler, conditions) tuple
+        is returned.
+        Otherwise, the file path of the saved scheduler is returned.
+    """
+    these_scheduler_references = await query_schedulers_in_window(desired_time, efd)
+    this_url = these_scheduler_references.url[0]
+    result = get_scheduler(this_url, destination)
+    return result
+
+
+async def get_scheduler_on_night(night, scheduler_index, destination, efd="usdf_efd"):
+    """Get a scheduler for a given night from the EFD.
+
+    Parameters
+    ----------
+    night : `str`
+        The night for which to get the scheduler.
+    scheduler_index : `int`, optional
+        The index of the scheduler to retrieve (default is 0).
+    destination : `str`, optional
+        The file path to save the scheduler to.
+        If None, the (scheduler, conditions) tuple is returned.
+    efd : `str`, optional
+        The EFD to query (default is 'usdf_efd').
+
+    Returns
+    -------
+    result : `np.ndarray`
+        The scheduler for the given night.
+    """
+    these_scheduler_references = await query_night_schedulers(night, efd=efd)
+    this_url = these_scheduler_references.url[scheduler_index]
+    result = get_scheduler(this_url, destination)
+    return result
+
+
+async def main():
+    parser = argparse.ArgumentParser(description="Query the EFD for a scheduler.")
```

Comment:
I don't think I understand this script .. does it returns the scheduler file name if you specify "destination" but simply returns a list of scheduler files if you do not? This seems like a mix of two things which are actually separate .. does it even need to be a "main" function?

Reply:
It can do either. I'll rework the command line to make it clearer.

Reply:
Done.